chore: pre-commit golangci-lint cannot analyze the go1.25 kukebuild module

## Context

Surfaced while implementing #656 (a kukebuild refactor). The `golangci-lint` pre-commit hook silently fails to lint the `cmd/kukebuild` module, so every kukebuild contributor hits a confusing panic at commit time and the module ships locally unlinted.

## Problem

`.pre-commit-config.yaml:38-42` pins `golangci-lint` at `v2.6.2`. That release's own module declares `go 1.24`, so the binary refuses any module targeting a newer Go — and `cmd/kukebuild/go.mod` is `go 1.25.5` (the deliberate module split from #662). The hook errors with:

```
panic: package requires newer Go version go1.25 (application built with go1.24)
```

or, with a freshly-installed binary:

```
can't load config: the Go language version (go1.24) used to build golangci-lint is lower than the targeted Go version (1.25.5)
```

Rebuilding/reinstalling golangci-lint v2.6.2 under go1.25 does **not** help — the go1.24 floor is baked into that release's own `go.mod`, not the build toolchain.

Net effect: a contributor touching `cmd/kukebuild` must `SKIP=golangci-lint git commit` to get past the hook, and the kukebuild module receives no lint coverage. (CI is unaffected — `.github/workflows/test.yaml` runs only `make test` + `make e2e`, no lint step — so this is purely local dev-loop friction, not a CI gate.)

## Proposal

Pick one:

1. **Bump golangci-lint** to a release whose module targets `go >= 1.25` so the hook can analyze both modules. Verify it still lints the root (`go 1.24.5`) module too.
2. **Scope the hook to the root module** and add a second hook entry (or a `local` hook) that runs golangci-lint from inside `cmd/kukebuild/` with a go1.25-capable binary, so kukebuild keeps lint coverage.
3. **Explicitly exclude `cmd/kukebuild/`** from the golangci-lint hook with an inline comment pointing at the module-split rationale, accepting no kukebuild lint coverage until option 1 is viable — at minimum this removes the confusing panic.

## Acceptance criteria

- [ ] `pre-commit run golangci-lint --files cmd/kukebuild/build.go` exits cleanly (lints or cleanly skips — no panic / version-floor error).
- [ ] The chosen approach is documented inline in `.pre-commit-config.yaml` referencing the #662 module split.
- [ ] Root-module lint coverage is unchanged.

## Out of scope

- Any change to CI (`.github/workflows/test.yaml` runs no lint step today; this issue does not propose adding one).

## References

- Surfaced during #656 / PR #691.
- Module split: #662 (`cmd/kukebuild/go.mod`, `go 1.25.5`) and #655 (`go.work`).
- Hook pin: `.pre-commit-config.yaml:38-42`.

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -39,6 +39,14 @@ repos:
     rev: v2.6.2
     hooks:
       - id: golangci-lint
+        # cmd/kukebuild lives in its own go 1.25 module (#662 module split).
+        # v2.6.2's binary is built with go 1.24; with go.work enabled, the
+        # workspace package loader pulls kukebuild into analysis and the
+        # type checker panics ("package requires newer Go version go1.25").
+        # Disable workspace mode so the hook only sees the root module
+        # (go 1.24.5). cmd/kukebuild stays unlinted by pre-commit until
+        # golangci-lint is bumped to a release built with go 1.25+ (#693).
+        entry: env GOWORK=off golangci-lint run --new-from-rev HEAD --fix
         args: ["./..."]
 
   - repo: https://github.com/pre-commit/mirrors-prettier

